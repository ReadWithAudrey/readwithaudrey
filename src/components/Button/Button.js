import React from 'react'
import PropTypes from 'prop-types'
import { StyledButton, HeaderButton, RegisterButton } from './Button.style'

const Button = props => {
  if (props.style === 'header') {
    return <HeaderButton>{props.children}</HeaderButton>
  } else if (props.style === 'register') {
    return <RegisterButton ontype={props.type}>{props.children}</RegisterButton>
  } else {
    return <StyledButton>{props.children}</StyledButton>
  }
}

Button.propTypes = {
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
  type: PropTypes.string,
}

export default Button
