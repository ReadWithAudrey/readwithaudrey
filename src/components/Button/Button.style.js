import styled from 'styled-components'

export const StyledButton = styled.button.attrs({
  className: 'bg-pink hover-bg-green white pa2 br2 w-50 h3 f3 montserrat',
})`
  border: none;
  display: block;
  a {
    color: #fff;
  }
`

export const HeaderButton = styled.button.attrs({
  className: 'bg-green hover-bg-pink pa2 mv2 white br2 w-100 f5 montserrat',
})`
  border: 2px solid white;
`

export const RegisterButton = styled.button.attrs({
  className:
    'bg-pink hover-bg-green white pa2 br2 w-40 w-20-ns h3 f3 mt4 mb4 center',
})`
  border: none;
  display: block;
  white-space: nowrap;
  min-width: 200px;
  max-width: 400px;

  a {
    color: #fff;
  }
`
