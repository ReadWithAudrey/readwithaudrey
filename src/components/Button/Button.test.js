import React from 'react'
import renderer from 'react-test-renderer'
import Button from './Button'

describe('Button component snapshot test', () =>
  it('renders correctly', () => {
    const tree = renderer.create(<Button />).toJSON()
    expect(tree).toMatchSnapshot()
  }))
