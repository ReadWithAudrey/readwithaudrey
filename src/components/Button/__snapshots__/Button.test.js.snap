// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component snapshot test renders correctly 1`] = `
<button
  className="sc-bdVaJa bg-pink hover-bg-orange white pa2 br2 w-50 h3 f3 montserrat hbxQQf"
/>
`;
